cng/ecdsa.go

```diff
@@ -237,10 +237,21 @@ func SignECDSA(priv *PrivateKeyECDSA, hash []byte) (r, s BigInt, err error) {
 
 // VerifyECDSA verifies the signature in r, s of hash using the public key, pub.
 func VerifyECDSA(pub *PublicKeyECDSA, hash []byte, r, s BigInt) bool {
-	if len(r) != len(s) {
+	// r and s might be shorter than size
+	// if the original big number contained leading zeros.
+	if len(r) > pub.size || len(s) > pub.size {
```

Comment:
I think the comment makes sense in the context of the PR but seems unclear without that... I'd suggest:

```suggestion
	// r and s might be shorter than size
	// if the original big number contained leading zeros,
	// but they must not be longer than the public key size.
	if len(r) > pub.size || len(s) > pub.size {
```